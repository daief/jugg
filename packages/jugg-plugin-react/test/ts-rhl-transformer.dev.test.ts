import * as ts from 'typescript';
import { join } from 'path';
import transformerFactory from '../src/ts-rhl-transformer/ts-rhl-transformer.dev';
import { readdirSync, statSync, readFileSync } from 'fs';
const printer = ts.createPrinter();
const FIXTURES_DIR = join(__dirname, '__fixtures__');

describe('ts-rhl-transformer', () => {
  describe('react components', () => {
    readdirSync(FIXTURES_DIR).forEach(fixtureName => {
      const fixtureFile = join(FIXTURES_DIR, fixtureName);

      if (statSync(fixtureFile).isFile()) {
        it(fixtureName, () => {
          const sourceCode = readFileSync(fixtureFile, 'utf-8');
          const source = ts.createSourceFile(fixtureName, sourceCode, ts.ScriptTarget.ES2016, true);
          const result = ts.transform(source, [transformerFactory()]);
          const transformedSourceFile = result.transformed[0];
          const resultCode = printer.printFile(transformedSourceFile);
          expect(resultCode).toMatchSnapshot();
        });
      }
    });
  });

  describe('copies arrow function body block onto hidden class methods', () => {
    const fixturesDir = join(FIXTURES_DIR, 'class-properties');

    readdirSync(fixturesDir).forEach(fixtureName => {
      const fixtureFile = join(fixturesDir, fixtureName);

      if (statSync(fixtureFile).isFile()) {
        it(fixtureName, () => {
          const sourceCode = readFileSync(fixtureFile, 'utf-8');
          const source = ts.createSourceFile(fixtureName, sourceCode, ts.ScriptTarget.ES2016, true);
          const result = ts.transform(source, [transformerFactory()]);
          const transformedSourceFile = result.transformed[0];
          const resultCode = printer.printFile(transformedSourceFile);
          expect(resultCode).toMatchSnapshot();
        });
      }
    });
  });
});
