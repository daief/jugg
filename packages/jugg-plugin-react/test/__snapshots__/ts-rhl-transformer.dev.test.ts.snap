// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ts-rhl-transformer copies arrow function body block onto hidden class methods arguments.js 1`] = `
"
(function () {
  var enterModule = require('react-hot-loader').enterModule;
  enterModule && enterModule(module);
}())
;


;
class Foo {
    bar = (a, b) => {
        arguments;
        return a(b);
    };
    __reactstandin__regenerateByEval(key, code) { this[key] = eval(code); }
}


;
(function () { 
var reactHotLoader = require('react-hot-loader').default;
var leaveModule = require('react-hot-loader').leaveModule;
if (!reactHotLoader) {
  return;
}
; reactHotLoader.register(Foo, \\"Foo\\", \\"arguments.js\\"); 
; leaveModule(module); }());


;
"
`;

exports[`ts-rhl-transformer copies arrow function body block onto hidden class methods arrow-function-in-constructor.js 1`] = `
"
(function () {
  var enterModule = require('react-hot-loader').enterModule;
  enterModule && enterModule(module);
}())
;


;
class Foo {
    constructor() {
        this.onClick = (e) => e.target.value;
    }
    __reactstandin__regenerateByEval(key, code) { this[key] = eval(code); }
}


;
(function () { 
var reactHotLoader = require('react-hot-loader').default;
var leaveModule = require('react-hot-loader').leaveModule;
if (!reactHotLoader) {
  return;
}
; reactHotLoader.register(Foo, \\"Foo\\", \\"arrow-function-in-constructor.js\\"); 
; leaveModule(module); }());


;
"
`;

exports[`ts-rhl-transformer copies arrow function body block onto hidden class methods async-functions.js 1`] = `
"
(function () {
  var enterModule = require('react-hot-loader').enterModule;
  enterModule && enterModule(module);
}())
;


;
class Foo {
    bar = async (a, b) => {
        return await a(b);
    };
    __reactstandin__regenerateByEval(key, code) { this[key] = eval(code); }
}


;
(function () { 
var reactHotLoader = require('react-hot-loader').default;
var leaveModule = require('react-hot-loader').leaveModule;
if (!reactHotLoader) {
  return;
}
; reactHotLoader.register(Foo, \\"Foo\\", \\"async-functions.js\\"); 
; leaveModule(module); }());


;
"
`;

exports[`ts-rhl-transformer copies arrow function body block onto hidden class methods async-functions-expression-body.js 1`] = `
"
(function () {
  var enterModule = require('react-hot-loader').enterModule;
  enterModule && enterModule(module);
}())
;


;
class Foo {
    bar = async (a, b) => await b(a);
    __reactstandin__regenerateByEval(key, code) { this[key] = eval(code); }
}


;
(function () { 
var reactHotLoader = require('react-hot-loader').default;
var leaveModule = require('react-hot-loader').leaveModule;
if (!reactHotLoader) {
  return;
}
; reactHotLoader.register(Foo, \\"Foo\\", \\"async-functions-expression-body.js\\"); 
; leaveModule(module); }());


;
"
`;

exports[`ts-rhl-transformer copies arrow function body block onto hidden class methods block-body.js 1`] = `
"
(function () {
  var enterModule = require('react-hot-loader').enterModule;
  enterModule && enterModule(module);
}())
;


;
class Foo {
    bar = (a, b) => {
        return a(b);
    };
    __reactstandin__regenerateByEval(key, code) { this[key] = eval(code); }
}


;
(function () { 
var reactHotLoader = require('react-hot-loader').default;
var leaveModule = require('react-hot-loader').leaveModule;
if (!reactHotLoader) {
  return;
}
; reactHotLoader.register(Foo, \\"Foo\\", \\"block-body.js\\"); 
; leaveModule(module); }());


;
"
`;

exports[`ts-rhl-transformer copies arrow function body block onto hidden class methods default-params.js 1`] = `
"
(function () {
  var enterModule = require('react-hot-loader').enterModule;
  enterModule && enterModule(module);
}())
;


;
class Foo {
    bar = (a = \\"foo\\") => {
        return \`\${a}bar\`;
    };
    __reactstandin__regenerateByEval(key, code) { this[key] = eval(code); }
}


;
(function () { 
var reactHotLoader = require('react-hot-loader').default;
var leaveModule = require('react-hot-loader').leaveModule;
if (!reactHotLoader) {
  return;
}
; reactHotLoader.register(Foo, \\"Foo\\", \\"default-params.js\\"); 
; leaveModule(module); }());


;
"
`;

exports[`ts-rhl-transformer copies arrow function body block onto hidden class methods destructured-params.js 1`] = `
"
(function () {
  var enterModule = require('react-hot-loader').enterModule;
  enterModule && enterModule(module);
}())
;


;
class Foo {
    bar = ({ a }, { b }) => {
        return \`\${a}\${b}\`;
    };
    __reactstandin__regenerateByEval(key, code) { this[key] = eval(code); }
}


;
(function () { 
var reactHotLoader = require('react-hot-loader').default;
var leaveModule = require('react-hot-loader').leaveModule;
if (!reactHotLoader) {
  return;
}
; reactHotLoader.register(Foo, \\"Foo\\", \\"destructured-params.js\\"); 
; leaveModule(module); }());


;
"
`;

exports[`ts-rhl-transformer copies arrow function body block onto hidden class methods expression-body.js 1`] = `
"
(function () {
  var enterModule = require('react-hot-loader').enterModule;
  enterModule && enterModule(module);
}())
;


;
class Foo {
    onClick = (e) => e.target.value;
    __reactstandin__regenerateByEval(key, code) { this[key] = eval(code); }
}


;
(function () { 
var reactHotLoader = require('react-hot-loader').default;
var leaveModule = require('react-hot-loader').leaveModule;
if (!reactHotLoader) {
  return;
}
; reactHotLoader.register(Foo, \\"Foo\\", \\"expression-body.js\\"); 
; leaveModule(module); }());


;
"
`;

exports[`ts-rhl-transformer copies arrow function body block onto hidden class methods nested-arguments.js 1`] = `
"
(function () {
  var enterModule = require('react-hot-loader').enterModule;
  enterModule && enterModule(module);
}())
;


;
class Foo {
    bar = (a, b) => {
        () => {
            arguments;
        };
        return a(b);
    };
    __reactstandin__regenerateByEval(key, code) { this[key] = eval(code); }
}


;
(function () { 
var reactHotLoader = require('react-hot-loader').default;
var leaveModule = require('react-hot-loader').leaveModule;
if (!reactHotLoader) {
  return;
}
; reactHotLoader.register(Foo, \\"Foo\\", \\"nested-arguments.js\\"); 
; leaveModule(module); }());


;
"
`;

exports[`ts-rhl-transformer copies arrow function body block onto hidden class methods nested-new.target.js 1`] = `
"
(function () {
  var enterModule = require('react-hot-loader').enterModule;
  enterModule && enterModule(module);
}())
;


;
class Foo {
    bar = (a, b) => {
        () => {
            new.target;
        };
        return a(b);
    };
    __reactstandin__regenerateByEval(key, code) { this[key] = eval(code); }
}


;
(function () { 
var reactHotLoader = require('react-hot-loader').default;
var leaveModule = require('react-hot-loader').leaveModule;
if (!reactHotLoader) {
  return;
}
; reactHotLoader.register(Foo, \\"Foo\\", \\"nested-new.target.js\\"); 
; leaveModule(module); }());


;
"
`;

exports[`ts-rhl-transformer copies arrow function body block onto hidden class methods new.target.js 1`] = `
"
(function () {
  var enterModule = require('react-hot-loader').enterModule;
  enterModule && enterModule(module);
}())
;


;
class Foo {
    bar = (a, b) => {
        new.target;
        return a(b);
    };
    __reactstandin__regenerateByEval(key, code) { this[key] = eval(code); }
}


;
(function () { 
var reactHotLoader = require('react-hot-loader').default;
var leaveModule = require('react-hot-loader').leaveModule;
if (!reactHotLoader) {
  return;
}
; reactHotLoader.register(Foo, \\"Foo\\", \\"new.target.js\\"); 
; leaveModule(module); }());


;
"
`;

exports[`ts-rhl-transformer copies arrow function body block onto hidden class methods not-a-function.js 1`] = `
"
(function () {
  var enterModule = require('react-hot-loader').enterModule;
  enterModule && enterModule(module);
}())
;


;
class Foo {
    bar = 2;
    __reactstandin__regenerateByEval(key, code) { this[key] = eval(code); }
}


;
(function () { 
var reactHotLoader = require('react-hot-loader').default;
var leaveModule = require('react-hot-loader').leaveModule;
if (!reactHotLoader) {
  return;
}
; reactHotLoader.register(Foo, \\"Foo\\", \\"not-a-function.js\\"); 
; leaveModule(module); }());


;
"
`;

exports[`ts-rhl-transformer copies arrow function body block onto hidden class methods not-an-arrow-function.js 1`] = `
"
(function () {
  var enterModule = require('react-hot-loader').enterModule;
  enterModule && enterModule(module);
}())
;


;
class Foo {
    bar = function (a, b) {
        return a(b);
    };
    __reactstandin__regenerateByEval(key, code) { this[key] = eval(code); }
}


;
(function () { 
var reactHotLoader = require('react-hot-loader').default;
var leaveModule = require('react-hot-loader').leaveModule;
if (!reactHotLoader) {
  return;
}
; reactHotLoader.register(Foo, \\"Foo\\", \\"not-an-arrow-function.js\\"); 
; leaveModule(module); }());


;
"
`;

exports[`ts-rhl-transformer copies arrow function body block onto hidden class methods only-constructor.js 1`] = `
"
(function () {
  var enterModule = require('react-hot-loader').enterModule;
  enterModule && enterModule(module);
}())
;


;
class Foo {
    constructor() {
    }
    __reactstandin__regenerateByEval(key, code) { this[key] = eval(code); }
}


;
(function () { 
var reactHotLoader = require('react-hot-loader').default;
var leaveModule = require('react-hot-loader').leaveModule;
if (!reactHotLoader) {
  return;
}
; reactHotLoader.register(Foo, \\"Foo\\", \\"only-constructor.js\\"); 
; leaveModule(module); }());


;
"
`;

exports[`ts-rhl-transformer copies arrow function body block onto hidden class methods same-name-as-class-method.js 1`] = `
"
(function () {
  var enterModule = require('react-hot-loader').enterModule;
  enterModule && enterModule(module);
}())
;


;
class Foo {
    bar = (a, b) => {
        return a(b);
    };
    bar() {
        return 2;
    }
    __reactstandin__regenerateByEval(key, code) { this[key] = eval(code); }
}


;
(function () { 
var reactHotLoader = require('react-hot-loader').default;
var leaveModule = require('react-hot-loader').leaveModule;
if (!reactHotLoader) {
  return;
}
; reactHotLoader.register(Foo, \\"Foo\\", \\"same-name-as-class-method.js\\"); 
; leaveModule(module); }());


;
"
`;

exports[`ts-rhl-transformer copies arrow function body block onto hidden class methods static-property.js 1`] = `
"
(function () {
  var enterModule = require('react-hot-loader').enterModule;
  enterModule && enterModule(module);
}())
;


;
class Foo {
    static bar = (a, b) => {
        return a(b);
    };
}


;
(function () { 
var reactHotLoader = require('react-hot-loader').default;
var leaveModule = require('react-hot-loader').leaveModule;
if (!reactHotLoader) {
  return;
}
; reactHotLoader.register(Foo, \\"Foo\\", \\"static-property.js\\"); 
; leaveModule(module); }());


;
"
`;

exports[`ts-rhl-transformer react components bindings.js 1`] = `
"
(function () {
  var enterModule = require('react-hot-loader').enterModule;
  enterModule && enterModule(module);
}())
;


;
import P, { Q } from \\"left-pad\\";
const A = 42;
function B() {
    function R() { }
    class S {
    }
    const T = 42;
}
export class C {
    U() {
        function V() {
            class W {
            }
        }
    }
    __reactstandin__regenerateByEval(key, code) { this[key] = eval(code); }
}
const D = class X {
};
let E = D;
var Y = require(\\"left-pad\\");
var { Z } = require(\\"left-pad\\");
const _default_1 = React.createClass({})
export default _default_1;


;
(function () { 
var reactHotLoader = require('react-hot-loader').default;
var leaveModule = require('react-hot-loader').leaveModule;
if (!reactHotLoader) {
  return;
}
; reactHotLoader.register(A, \\"A\\", \\"bindings.js\\"); 
; reactHotLoader.register(B, \\"B\\", \\"bindings.js\\"); 
; reactHotLoader.register(C, \\"C\\", \\"bindings.js\\"); 
; reactHotLoader.register(D, \\"D\\", \\"bindings.js\\"); 
; reactHotLoader.register(E, \\"E\\", \\"bindings.js\\"); 
; reactHotLoader.register(_default_1, \\"default\\", \\"bindings.js\\"); 
; leaveModule(module); }());


;
"
`;

exports[`ts-rhl-transformer react components counter.js 1`] = `
"
(function () {
  var enterModule = require('react-hot-loader').enterModule;
  enterModule && enterModule(module);
}())
;


;
class Counter {
}
const _default_1 = () => <div><Counter /></div>
export default _default_1;


;
(function () { 
var reactHotLoader = require('react-hot-loader').default;
var leaveModule = require('react-hot-loader').leaveModule;
if (!reactHotLoader) {
  return;
}
; reactHotLoader.register(Counter, \\"Counter\\", \\"counter.js\\"); 
; reactHotLoader.register(_default_1, \\"default\\", \\"counter.js\\"); 
; leaveModule(module); }());


;
"
`;

exports[`ts-rhl-transformer react components export-default-anonymous-class.js 1`] = `
"
(function () {
  var enterModule = require('react-hot-loader').enterModule;
  enterModule && enterModule(module);
}())
;


;
const _default_1 = class {
    a() { }
    __reactstandin__regenerateByEval(key, code) { this[key] = eval(code); }
}
export default _default_1;


;
(function () { 
var reactHotLoader = require('react-hot-loader').default;
var leaveModule = require('react-hot-loader').leaveModule;
if (!reactHotLoader) {
  return;
}
; reactHotLoader.register(_default_1, \\"default\\", \\"export-default-anonymous-class.js\\"); 
; leaveModule(module); }());


;
"
`;

exports[`ts-rhl-transformer react components export-default-assignment-arrow-func.js 1`] = `
"
(function () {
  var enterModule = require('react-hot-loader').enterModule;
  enterModule && enterModule(module);
}())
;


;
const _default_1 = () => console.log('export-default-assignment-arrow-func')
export default _default_1;


;
(function () { 
var reactHotLoader = require('react-hot-loader').default;
var leaveModule = require('react-hot-loader').leaveModule;
if (!reactHotLoader) {
  return;
}
; reactHotLoader.register(_default_1, \\"default\\", \\"export-default-assignment-arrow-func.js\\"); 
; leaveModule(module); }());


;
"
`;

exports[`ts-rhl-transformer react components export-default-class.js 1`] = `
"
(function () {
  var enterModule = require('react-hot-loader').enterModule;
  enterModule && enterModule(module);
}())
;


;


;
(function () { 
var reactHotLoader = require('react-hot-loader').default;
var leaveModule = require('react-hot-loader').leaveModule;
if (!reactHotLoader) {
  return;
}
; reactHotLoader.register(a, \\"a\\", \\"export-default-class.js\\"); 
; leaveModule(module); }());


;
"
`;

exports[`ts-rhl-transformer react components name-clash.js 1`] = `
"
(function () {
  var enterModule = require('react-hot-loader').enterModule;
  enterModule && enterModule(module);
}())
;


;
const _default = 10;
const _default_1 = 42
export default _default_1;


;
(function () { 
var reactHotLoader = require('react-hot-loader').default;
var leaveModule = require('react-hot-loader').leaveModule;
if (!reactHotLoader) {
  return;
}
; reactHotLoader.register(_default, \\"_default\\", \\"name-clash.js\\"); 
; reactHotLoader.register(_default_1, \\"default\\", \\"name-clash.js\\"); 
; leaveModule(module); }());


;
"
`;

exports[`ts-rhl-transformer react components rhl-issye-246.js 1`] = `
"
(function () {
  var enterModule = require('react-hot-loader').enterModule;
  enterModule && enterModule(module);
}())
;


;
export function spread(...args) {
    return args.push(1);
}


;
(function () { 
var reactHotLoader = require('react-hot-loader').default;
var leaveModule = require('react-hot-loader').leaveModule;
if (!reactHotLoader) {
  return;
}
; reactHotLoader.register(spread, \\"spread\\", \\"rhl-issye-246.js\\"); 
; leaveModule(module); }());


;
"
`;
